module/apmawssdkgo/session.go

```diff
@@ -189,6 +189,12 @@ func send(req *request.Request) {
 }
 
 func complete(req *request.Request) {
+	spanSubtype := req.ClientInfo.ServiceName
+	_, ok := serviceTypeMap[spanSubtype]
```

Comment:
nit: could you move the check to a function similar to 
```

func supportedRequest(req *request.Request) bool {
  spanSubtype := req.ClientInfo.ServiceName
  _, ok := serviceTypeMap[spanSubtype]
  return ok
}
```
and share between the `complete` and `build` method?